### PR TITLE
Use char overloads to Join/Index strings

### PR DESCRIPTION
**Pull request by martincostello**
# Use char overloads to Join/Index strings

Use overloads that take a `char` for single-character strings.

## Description

Use overloads that take a `char` for single-character strings where relevant with `string.Join()` or `{Last}IndexOf()` to resolve a number of analyser suggestions.

Also fixes a few IDE0055 diagnostics.

### CODE REVIEW

**Comment by amcasey** [comment]
CI failure looks unrelated - that analyzer has been failing in a lot of runs.

File: src/Http/Http/src/BindingAddress.cs [hunk 1]
```diff
@@ -179,17 +179,17 @@ public static BindingAddress Parse(string address)
                 }
             }
 
-            pathDelimiterStart = address.IndexOf(":", schemeDelimiterEnd + unixPipeHostPrefixLength, StringComparison.Ordinal);
```
Comment (amcasey) [inline src/Http/Http/src/BindingAddress.cs:182]: Not your fault, but it seems very weird that the char version doesn't (AFAICT) accept a `StringComparison`.  I'm assuming the default is `Ordinal`?

Comment (martincostello) [reply src/Http/Http/src/BindingAddress.cs:182]: I'll double-check this one locally in an IDE.

Comment (martincostello) [reply src/Http/Http/src/BindingAddress.cs:182]: I couldn't find one a `char, int, StringComparison` overload.

The IDE code-fixer also suggests exactly what's here:

image

I'm gonna _guess_ that it does an ordinal check anyway and in this case as it's punctuation it's moot.

Comment (amcasey) [reply src/Http/Http/src/BindingAddress.cs:182]: Sorry, I should have been clearer - there isn't a better overload you could have called.  I don't know why that is, but you don't need to do anything about it.

I just confirmed in the source that the default comparison is `Ordinal` so the change looks correct (if, sadly, less explicit).

**Review by amcasey — APPROVED** [review]

**Pull request merged**